### ISSUE TITLE
Reproduction of the DebiasPL

### ISSUE THREAD

**Issue by seekingup** [microsoft/Semi-supervised-learning#76]
Hi, when I reproduced the result of FixMatch and DebiasPL (the imbalanced algorithm), I found something wrong. It may be a bug.

If I understand correctly, if I set the debiaspl_tau=0, then the training result of fixmatch_debiaspl should be the same with fixmatch.
Thus, I run the two config:
1. DebiasPL_0: `config/classic_cv_imb/fixmatch_debiaspl/fixmatch_debiaspl_cifar10_lb1500_100_ulb3000_100_0.yaml`, and add `debiaspl_tau=0` into the config.
2. FixMatch: `config/classic_cv_imb/fixmatch/fixmatch_cifar10_lb1500_100_ulb3000_100_0.yaml`

However, their performance are different. The FixMatch has a significant accuracy drop in the late phase of training, which DebiasPL_0  does not. Please see the curve below.

The last iteration accuracy and best accuracy are 77.46 / 77.95 (DebiasPL_0) and 74.48 / 77.94 (FixMatch).
<img width="730" alt="image" src="https://user-images.githubusercontent.com/40528686/220232250-15c73402-2d6e-4d20-b8ac-30f0a213faae.png">
<img width="732" alt="image" src="https://user-images.githubusercontent.com/40528686/220232646-d9ab99b7-5419-4726-9f77-3fb24dc75048.png">

I ran the experiments multiple times and found no clues. Is there a bug or there is something I missed?

**Comment by Hhhhhhao** [comment]
Good catch. Your understanding is correct. Setting tau as 0 would make DebiasPL the same as FixMatch in our implementation. 

I think the difference lies in the learning rate scheduler. As you can see the best accuracy of the experiments you ran is similar (also similar to our reported results). The problem is, these imbalanced algorithms do not use any schedulers (in their paper setup and open-source code), so I set the scheduler as None in ImbAlgorithm Base class at here: https://github.com/microsoft/Semi-supervised-learning/blob/b979a23a1065f4d427350534a75ba1beb8c2afe9/semilearn/core/algorithmbase.py#L531
This means DebiasPL with tau=0 would also not use any scheduler.

But if you run FixMatch, it falls into original Algorithm Base class, which will use the cosine scheduler, causing performance drop at later iterations.

**Comment by seekingup** [comment]
Thanks, after checking the training logs, I found the differences. 
However, I also checked the paper and open-source code of DebiasPL. It seems DebiasPL uses cosine scheduler as well. Could you kindly give me some sources of why imbalanced algorithms do not use any schedulers?

**Comment by Hhhhhhao** [comment]
You can check ABC paper: https://arxiv.org/pdf/2110.10368.pdf.
I believed other papers like DARP and DASO also mentioned no learning rate scheduler in their paper.

**Comment by seekingup** [comment]
thank you so much!
